src/ui/common/src/utils/workflows.tsx

```diff
@@ -212,13 +213,21 @@ export function getWorkflowEngineTypes(dag: WorkflowDag): EngineType[] {
         .filter((t) => !!t) ?? []
     )
   );
+
+  return reduceEngineTypes(dagType, allOpTypes);
+}
+
+export function reduceEngineTypes(
+  dagType: EngineType,
+  opTypes: EngineType[]
+): EngineType[] {
   if (
     dagType === EngineType.Aqueduct &&
-    allOpTypes.length > 0 &&
-    allOpTypes.every((t) => t !== EngineType.Aqueduct)
+    opTypes.length > 0 &&
```

Comment:
Just want to make sure that I'm understanding this right, but this function first filters out all non Aqueduct engines and makes sure that they are all unique (new Set part on line 232), then puts them into an array?

So the final result here is an Array of unique engine types excluding Aqueduct?

Reply:
yes, the conditioning ensures we show only operator types if no operator runs on aqueduct. This is possible as we are still orchestrating using aqueduct (and the dag type is Aq)